packages/react-shadow/src/root.tsx

```diff
@@ -24,10 +32,33 @@ const ReactComponentsWrapper: React.FC<{
         return;
       }
 
-      root.adoptedStyleSheets = [
-        ...root.adoptedStyleSheets,
-        ...renderer.adoptedStyleSheets,
-      ];
```

Comment:
TL;DR This change should be ported to Griffel, the relevant place is `findInsertionPoint()`: https://github.com/microsoft/griffel/blob/f134bc6056599a78cf27b71dec854224a162ee22/packages/shadow-dom/src/findInsertionPoint.ts#L9

----

As I recall, this effect is solving the issue of syncing sheets from `react-shadow` and Griffel as the logic in `react-shadow` will simply erase them:

https://github.com/Wildhoney/ReactShadow/blob/1432b818cf0ba96b8a86920d4b426eb41911358c/src/core/index.js#L60-L61

However, now that effect will re-create also `root`, so new Griffel renderer will be created. I created a PR to remove this logic in the first place (#61).